lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/sys.rb

```diff
@@ -269,10 +317,46 @@ def cmd_shell(*args)
       path = client.fs.file.expand_path("%COMSPEC%")
       # If that failed for whatever reason, guess it's unix
       path = (path and not path.empty?) ? path : "/bin/sh"
+      if use_alt && path == "/bin/sh"
+        return true if alt_shell(alt_path)
+      end
       cmd_execute("-f", path, "-c", "-i")
     end
   end
 
+  #
+  # Spawn an alternative shell
+  #
+  def alt_shell(sh_path)
+    sh_path = client.fs.file.exist?(sh_path) ? sh_path : '/bin/sh'
+
+    # Python Meterpreter calls pty.openpty() - No need for other methods
+    if client.session_type =~ /python/
+      cmd_execute('-if', sh_path)
+      return true
+    end
+
+    # Select method for spawning PTY Shell based on availability on the target.
+    path = nil
+    path = '/usr/bin/socat'   if client.fs.file.exist?('/usr/bin/socat')
+    path = '/usr/bin/script'  if client.fs.file.exist?('/usr/bin/script')  && !path
+    path = '/usr/bin/python'  if client.fs.file.exist?('/usr/bin/python')  && !path
+    path = '/usr/bin/python2' if client.fs.file.exist?('/usr/bin/python2') && !path
+    path = '/usr/bin/python3' if client.fs.file.exist?('/usr/bin/python3') && !path
+    path = '/usr/bin/expect'  if client.fs.file.exist?('/usr/bin/expect')  && !path
+    return false unless path
+
+    # Commands for methods - "export TERM=xterm" provides colors, "clear" command, etc. as available on the target.
+    cmd = "export TERM=xterm; #{path} - exec:'#{sh_path} -li',pty,stderr,setsid,sigint,sane" if path =~ /socat/
```

Comment:
Consider simply prefixing `TERM=xterm` or using `env(1)` (potentially more portable).

If you require an exported variable in the shell's environment (doesn't look like you do), you should use the portable syntax `TERM=xterm; export TERM`.

http://wiki.bash-hackers.org/scripting/nonportable